art/estimators/classification/pytorch.py

```diff
@@ -847,11 +933,12 @@ def get_activations(
 
         else:  # pragma: no cover
             raise TypeError("Layer must be of type str or int")
-
         if framework:
             if isinstance(x, torch.Tensor):
                 return self._model(x)[layer_index]
-            return self._model(torch.from_numpy(x).to(self._device))[layer_index]
+            x_ = torch.from_numpy(x_preprocessed)
+            self._model(x_.to(self._device))
+            return x_, self._model._features[layer]
```

Comment:
Are these changes required?

Reply:
since, we are using hooks, we first need to pass the input through the model, which creates evaluation hooks at every layer and then we index that layer to get output. We can probably remove x_ (that is not required)